### ISSUE TITLE
Not building on ghc 9.2.2

### ISSUE THREAD

**Issue by HugoPeters1024** [well-typed/large-records#113]
I tried to use the library in a project using ghc 9.2.2 (which has a compatable base version according to the cabal file) but I get the following error

```sh
Configuring library for large-records-0.2.1.0..
Preprocessing library for large-records-0.2.1.0..
Building library for large-records-0.2.1.0..
[ 1 of 12] Compiling Data.Record.Internal.GHC.Shim ( src/Data/Record/Internal/GHC/Shim.hs, dist/build/Data/Record/Internal/GHC/Shim.o, dist/build/Data/Record/Internal/GHC/Shim.dyn_o )

src/Data/Record/Internal/GHC/Shim.hs:101:25: error:
    Module ‘GHC.Types.Basic’ does not export ‘SourceText’
    |
101 | import GHC.Types.Basic (SourceText(NoSourceText))
    |                         ^^^^^^^^^^^^^^^^^^^^^^^^

src/Data/Record/Internal/GHC/Shim.hs:103:57: error:
    Module ‘GHC.Utils.Error’ does not export ‘mkErrMsg’
    |
103 | import GHC.Utils.Error (Severity(SevError, SevWarning), mkErrMsg, mkWarnMsg)
    | 
```

Likely these import have moved to different modules and I did some looking around to find a quick fix but a lot of the documentation surrounding these imports appears to not exist for some reason. Perhaps there is an obvious fix I am not aware of?

**Comment by HugoPeters1024** [comment]
Thanks for the response, I'll try to give it a shot for a PR

**Comment by HugoPeters1024** [comment]
With some version macros I was able to resolve the initial errors but it brought to light that GHC has had more drastic changes:

```sh
src/Data/Record/Internal/GHC/Shim.hs:123:29: error:
    • Couldn't match type ‘SrcSpan’
                     with ‘SrcSpanAnn' (EpAnn AnnListItem)’
      Expected: LImportDecl GhcPs
        Actual: Located (ImportDecl GhcPs)
    • In the expression:
        noLoc
          $ ImportDecl
              {ideclExt = defExt, ideclSourceSrc = NoSourceText,
               ideclName = noLoc name, ideclPkgQual = Nothing, ideclSafe = False,
               ideclImplicit = False, ideclAs = Nothing, ideclHiding = Nothing,
               ideclQualified = if qualified then QualifiedPre else NotQualified,
               ideclSource = NotBoot}
      In an equation for ‘importDecl’:
          importDecl name qualified
            = noLoc
                $ ImportDecl
                    {ideclExt = defExt, ideclSourceSrc = NoSourceText,
                     ideclName = noLoc name, ideclPkgQual = Nothing, ideclSafe = False,
                     ideclImplicit = False, ideclAs = Nothing, ideclHiding = Nothing,
                     ideclQualified = if qualified then QualifiedPre else NotQualified,
                     ideclSource = NotBoot}
    |
123 | importDecl name qualified = noLoc $ ImportDecl {
    |                             ^^^^^^^^^^^^^^^^^^^^...
```

...and many more such errors. This is where it goes above my paycheck I am afraid.

**Comment by edsko** [comment]
Oh the joys of supporting multiple ghc versions are without bound! :-) Thanks for taking a look. I will try to take a look myself at some point but I don't know at this point when I will get to that :/

**Comment by phadej** [comment]
I made revisions to large-records and large-anon:
- https://hackage.haskell.org/package/large-anon-0.1.0.0/revisions/
- https://hackage.haskell.org/package/large-records-0.2.1.0/revisions/
- https://hackage.haskell.org/package/large-records-0.2.0.0

> (see the "tested with" field in the cabal file

`cabal-install`s solver looks into `build-depends` only, `tested-with` is only informative field.

**Comment by edsko** [comment]
This is resolved in #117 . @phadej , what is left, just a release now I guess?

**Comment by phadej** [comment]
> This is resolved in #117 . @phadej , what is left, just a release now I guess?

AFAICT, yeah.

**Comment by edsko** [comment]
Released as version 0.3 https://hackage.haskell.org/package/large-records-0.3 .
